services/common/rs_server_common/data_retrieval/eodag_provider.py

```diff
@@ -0,0 +1,80 @@
+"""EODAG Provider."""
+from pathlib import Path
+
+from eodag import EODataAccessGateway, EOProduct
+
+from .provider import CreateProviderFailed, Product, Provider, TimeRange
+
+
+class EodagProvider(Provider):
+    """An EODAG provider.
+
+    It uses EODAG to provide data from external sources.
+    """
+
+    def __init__(self, config_file: str, provider: str):
+        """Create a EODAG provider.
+
+        :param config: the eodag configuration
+        """
+        self.provider: str = provider
+        self.client: EODataAccessGateway = self.init_eodag_client(config_file)
+        self.client.set_preferred_provider(self.provider)
+
+    def init_eodag_client(self, config_file: Path):
+        """Initialize the eodag client.
+
+        The EODAG client is initialized for the given provider.
+
+        :param config_file: the path to the eodag configuration file
+        :return: the initialized eodag client
+        """
+        try:
+            return EODataAccessGateway(config_file.as_posix())
+        except Exception as e:
+            raise CreateProviderFailed(f"Can't initialize {self.provider} provider") from e
+
+    def _specific_search(self, between: TimeRange) -> dict[str, Product]:
```

Comment:
you don't add some unit tests for the EodagProvider.search()

Reply:
This concept is already tested in endpoint, tests/test_cadip_list_endpoint.py where the eodag search function is tested, since _specific_search is just an wrapper, maybe we can conclude that all tests cases for this are covered in tests/test_cadip_list_endpoint.py.

Reply:
Test cases are not unit tests.
If we want to have some flexibility and maintainability, we need both :
* unit tests to verify local functionalities on a fine grain,
* integration tests to verify the interface between objects
* test cases to verify the overall functionality covers the user needs

Your test case is very useful to verify we cover the user need
but we can't rely on him for day to day work.
Otherwise, we will be slower and slower developing new stories
(or we will not run it so we will not see issues).
Fast unit tests can be run several time a day without losing time.
They are really useful to ensure non regression on the software.

Reply:
That's my personal point of view.
Maybe, it's not the one we want to follow on this project.